manifests/vhost.pp

```diff
@@ -2707,7 +2711,64 @@
   # - $ssl_openssl_conf_cmd
   # - $ssl_stapling
   # - $apache_version
-  if $ssl {
+  if $ssl and $ensure == 'present' {
+    if $ssl_reload_on_change {
```

Comment:
No it can't - see line 2788, where the concat::fragment is located, which must be handled even if the reload feature is not used. Maybe it is clearer when i shift the concat::fragment higher and the reload stuff after that?

Reply:
I do think that would be easier to follow. Long deeply nested structures aren't that clear. Perhaps it's also not needed if https://github.com/puppetlabs/puppetlabs-apache/pull/2157#discussion_r663036253 works and can be implemented.

Reply:
Yes, you are right, but i will do the shift anyway even with the proposal of the other comment, because i find that also better.